src/molinillo_solver.cr

```diff
@@ -47,7 +47,38 @@ module Shards
       end
     end
 
+    private def prefetch_local_caches(lock_index, deps)
+      count = 0
+      active = 0
+      ch = Channel(Exception?).new(deps.size + 1)
+      deps.each do |dep|
+        next unless lock = lock_index[dep.name]?
+        next unless dep.matches?(lock.version)
+        count += 1
+        active += 1
+        while active > Shards.parallel_fetch
+          sleep 0.1
+        end
+        spawn do
+          begin
+            dep.resolver.update_local_cache
+            ch.send(nil)
+            active -= 1
+          rescue ex : Exception
+            ch.send(ex)
+          end
+        end
+      end
+
+      count.times do
+        obj = ch.receive
+        raise obj if obj.is_a? Exception
```

Comment:
This will raise only 1st `Exception`.
How about combining all found ones into a separate exception class and raising that instead?

Reply:
Doable but would feel like overkill to me?
When one of the fetches raises (usually due to a network error) then we have to abort anyway.

Not opposed to changing it, but skeptical it's worth the addtl code complexity.

Reply:
I think it's okay to leave it like that. Showing all exceptions at once is useful when they have *different* causes (for example, misspelled names of two dependencies), but not when they have the same cause (for example, not network connection).

It's probably more likely that a global error affects multiple tracks than two independent errors showing up at the same time. The latter can happen of course, but then you just need to fix them sequentially. I think that's better than repeating messages for the same error.
This is in line with the current behaviour where sequential resolving fails at the first error as well.

We can consider further improvements, for example collecting all errors and then deciding based on the type of errors, which ones should be printed. But that's for a follow-up and IMO not a high priority.